EntryManager.cpp

```diff
@@ -0,0 +1,90 @@
+#include <iostream>
+#include <string>
+#include <cctype> // used for the isalpha() & isspace() functions
+#include <windows.h> // used for the Sleep() function
+#include <pqxx/pqxx>
+#include "EntryManager.h"
+#include "sha256.h"
+#include "extern.h"
+
+using namespace std;
+
+void EntryManager::registerLibraryMember() {
+	string firstName, lastName, email, passHash;
+
+	cout << "--------------------- Registration ---------------------\n\n";
+	cout << "We're so glad you want to join us! We just need a few details from you to create an account!\n\n";
+	
+	firstName = obtainPII("first name");
+	lastName = obtainPII("last name");
+	email = obtainPII("email address");
+	passHash = createPassword();
+	
+	addNewLibraryMemberToDB(firstName, lastName, email, passHash);
+	completeLibraryMemberRegistration();
+}
+
+string EntryManager::obtainPII(string infoType) { // PII = personally identifiable information
+	string userInput;
+	cout << "What is your " << infoType << "?\n";
+	cout << "Enter your " << infoType << " here: "; getline(cin, userInput);
+	cout << endl;
+	checkPII(userInput, infoType);
+	return userInput;
+}
+
+void EntryManager::checkPII(string &userInput, string infoType) {
+	if (userInput == "") {
+		cout << "Invald input! Please try again!\n\n";
+		obtainPII(infoType);
+	}
+	
+	if (infoType == "first name" || infoType == "last name") {
+		for (int i = 0; i < userInput.length(); i++) {
+			if (userInput[0]) // ensures first and last names are capitalized
+				userInput[0] = toupper(userInput[0]);
+			
+			if (isalpha(userInput[i]) || userInput[i]) // used to prevent special chars and nums from being entered in names
+				continue;
+			else {
+				cout << "Invald input! Please try again!\n\n";
+				obtainPII(infoType);
+			}
+		}
+	}
+}
+
+string EntryManager::createPassword() {
+	string inputtedPassword;
+	cout << "Please provide a suitable password...\n";
+	cout << "Enter your password here: "; getline(cin, inputtedPassword);
```

Comment:
- A re-enter password field for user confirmation would be a nice feature to have